src/qt/veil/miningwidget.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2019 The Veil developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/veil/miningwidget.h>
+#include <qt/veil/forms/ui_miningwidget.h>
+
+// SOME OF THESE NEED TO BE DELETED
+#include <qt/guiconstants.h>
+#include <qt/guiutil.h>
+#include <qt/optionsmodel.h>
+#include <qt/platformstyle.h>
+#include <interfaces/node.h>
+#include <qt/walletview.h>
+#include <qt/walletmodel.h>
+#include <key_io.h>
+#include <wallet/wallet.h>
+#include <QGraphicsOpacityEffect>
+#include <QPropertyAnimation>
+#include <QPixmap>
+#include <QIcon>
+#include <QClipboard>
+#include <QMimeData>
+#include <QImage>
+#include <QLabel>
+#include <QPainter>
+#include <qt/veil/qtutils.h>
+#include <miner.h>
+
+MiningWidget::MiningWidget(QWidget *parent, WalletView* walletView) :
+    QWidget(parent),
+    ui(new Ui::MiningWidget),
+    mainWindow(walletView)
+{
+    ui->setupUi(this);
+    ui->title->setProperty("cssClass" , "title");
+
+    // SET THE NUMBER OF THREADS AND ALGORITHM HERE
+    mineOn = GenerateActive();
+
+    ui->cmbAlgoSelect->addItems({"RandomX","ProgPow","SHA256D"});
+    ui->cmbAlgoSelect->setCurrentIndex(GetMiningAlgorithm());
+
+    ui->lblMaxThreadsAvailable->setText(QString::number(GetNumCores() - 1));
+    
+
+    connect(ui->btnUpdateAlgo, SIGNAL(clicked()), this, SLOT(onUpdateAlgorithm()));
+    connect(ui->btnAllThreads, SIGNAL(clicked()), this, SLOT(onUseMaxThreads()));
+    connect(ui->btnActiveMine, SIGNAL(clicked()), this, SLOT(onToggleMiningActive()));
+
+}
+
+void MiningWidget::setMineActiveTxt(bool mineActive){
+    ui->lblMineActive->setText(mineActive ? "Mining Active" : "Mining Stopped");
+    ui->lblMineActive->setStyleSheet(mineActive ? "QLabel{color:green;}" : "QLabel{color:red;}");
+    ui->btnActiveMine->setText(mineActive ? "Stop Mining" : "Start Mining");
+}
+
+void MiningWidget::setWalletModel(WalletModel *model){
+    this->walletModel = model;
+
+    connect(model, SIGNAL(updateMiningFields()), this, SLOT(updateMiningFields()));
+}
+
+MiningWidget::~MiningWidget()
+{
+    delete ui;
+}
+
+void MiningWidget::onUpdateAlgorithm() {
```

Comment:
NIT.  I know it's a copied and edited file, but be sure to try and clean up a new file to a single style.
```
void module::method(params){
````
```
void module::method(params) {
````
```
void module::method(params)
{
````